src/api/app/controllers/webui/package_controller.rb

```diff
@@ -703,19 +703,14 @@ def trigger_rebuild
     # When we're in a linked project, the package's project points to some other
     # project, not the one we're triggering the build from.
     # Here we detect that, and if so, we authorize against the linked project.
-    if @project != @package.project
-      authorize @project, :update?
-    else
-      authorize @package, :update?
-    end
-
-    allowed_params = [:project, :package, :repository, :arch]
+    rebuild_trigger = PackageControllerService::RebuildTrigger.new(package: @package, project: @project, params: params)
+    authorize rebuild_trigger.fetch_policy, :update?
```

Comment:
I would like to move the policy check to the service, but because of the `verify_authorized` https://github.com/varvet/pundit/blob/6be4621538f916124f76b57460c8cc46597a3b6a/lib/pundit.rb#L192, it isn't  working

Reply:
You know you can skip this right? Just checking...

Reply:
for sure I know that we can skip it, but should I?

Reply:
Good question. We can talk if you want to!

Reply:
postponed to later